### PR TITLE
Update cpu relax functions

### PR DESCRIPTION
**Pull request by AGSaidi**
Update UT_RELAX_CPU to match upstream changes in MySQL
Fix cpu_relax and add RELAX_IS_ISB

### CODE REVIEW

File: ext/linux/include/lk_atomics.h [hunk 1]
```diff
@@ -251,7 +251,9 @@ static inline uint16_t xchg_release16(uint16_t *ptr, uint16_t val) {
 static inline void cpu_relax (void) {
 #if defined(__x86_64__)
 	asm volatile ("pause" : : : "memory" );
-#elif defined (__arch64__)
+#elif defined (__aarch64__) && defined(RELAX_IS_ISB)
```
Comment (rpgolshan) [inline ext/linux/include/lk_atomics.h:254]: When do you recommend a user enabling this over the default? Can you add a define in the Makefile?

File: ext/mysql/include/ut_atomics.h [hunk 2]
```diff
@@ -21,10 +21,10 @@ this program; if not, write to the Free Software Foundation, Inc.,
 
 #if defined(__x86_64__)
 #define UT_RELAX_CPU() asm volatile ("rep; nop")
-#elif defined(__AARCH64__)
+#elif defined(__aarch64__)
 // Theoretically we could emit a yield here but MySQL doesn't do it
 // and most ARM cores are likely to NOP it anyway
-#define UT_RELAX_CPU() asm volatile ("":::"memory")
+#define UT_RELAX_CPU() asm volatile ("isb":::"memory")
```
Comment (rpgolshan) [inline ext/mysql/include/ut_atomics.h:27]: This is fine.

File: ext/linux/include/lk_atomics.h (hunk 1 above)
Comment (AGSaidi) [reply ext/linux/include/lk_atomics.h:254]: Without the define, this behaves like the kernel does today. I believe the define improves the performance of the kernel locks but it's not what is committed today.

Comment (rpgolshan) [reply ext/linux/include/lk_atomics.h:254]: Ok so if the kernel changes in the future we just need to expose this define.

**Pull request merged**